src/java/org/jivesoftware/openfire/muc/spi/MultiUserChatServiceImpl.java

```diff
@@ -402,6 +416,21 @@ else if ("urn:xmpp:ping".equals(namespace)) {
             router.route( IQ.createResultIQ(iq) );
         }
         else {
+            IQHandler h = this.iqHandlers.get(namespace);
+            if (h != null) {
+                try {
+                    IQ reply = h.handleIQ(iq);
+                    if (reply != null) {
+                        router.route(reply);
+                    }
+                } catch (UnauthorizedException e) {
+                    IQ reply = IQ.createResultIQ(iq);
+                    reply.setType(IQ.Type.error);
+                    reply.setError(PacketError.Condition.service_unavailable);

```

Comment:
Ah, yes. It's the default error we should generally return, I think, since it's equivalent to an unknown namespace. A bunch of XEPs specify to return this in Unauthorized cases, too - not least because a private, unlisted room might otherwise unwittingly reveal its presence, for example.